sparkly/testing.py

```diff
@@ -172,40 +176,57 @@ def assertDataFrameEqual(self,
+                    # ordering parameters
                     ignore_order=False,
-                    max_recursion_depth=None,
-                    places=None,
-                    significant=False,
-                    equal_nan=True):
+                    ignore_order_depth=None,
+                    # float comparison parameters
+                    atol=0,
+                    rtol=1e-07,
+                    equal_nan=True,
+                    # DataType comparison parameters
+                    ignore_nullability=False):
         """Assert equal on steroids.
 
         Extend this classic function signature to work better with
-        comparisons involving dataframes, rows, dictionaries and lists by
-        ignoring the order of mutable nested structures recursively and
-        rounding up floats when necessary.
+        comparisons involving rows, datatypes, dictionaries, lists and
+        floats by:
+            - ignoring the order of lists and datatypes recursively,
+            - comparing floats within a given tolerance,
+            - assuming NaNs are equal,
+            - ignoring the nullability requirements of datatypes (since
+              Spark can be inaccurate when inferring it),
+            - providing better diffs for rows and datatypes.
+
+        Float comparisons are inspired by NumPy's assert_allclose. The
+        main formula used is | float1 - float2 | <= atol + rtol * float2.
 
         Args:
-            first, second, msg: see unittest.TestCase.assertCountEqual
-            ignore_order (bool|False): ignore the order in all structures
-                that are inherently ordered (e.g., lists, mock calls).
-            max_recursion_depth (int|None): if ignore_order is true, do
-                ignore order up to this nesting level (exclusive). Setting
-                this to 0 is same to setting ignore_order=False, 1 means
-                ignore order only at the top level and so on. Default is
-                ignore order arbitrarily deep.
-            places (int|None): Decimal or significant digits to consider
-                during comparisons of floats.
-            significant (bool|False): If set, interpret places as # of
-                significant digits instead of decimal.
+            first, second, msg: see unittest.TestCase.assertEqual
+            ignore_order (bool|False): ignore the order in lists and
+                datatypes (rows, dicts are inherently orderless).
+            ignore_order_depth (int|None): if ignore_order is true, do
+                ignore order up to this level of nested lists or
+                datatypes (exclusive). Setting this to 0 or None means
+                ignore order infinitely, 1 means ignore order only at the
+                top level, 2 will ignore order within lists of lists and
+                so on. Default is ignore order arbitrarily deep.
+            atol (int, float|0): Absolute tolerance in float comparisons.
```

Comment:
I prefer to use a shortcut that is standard when talking about tolerances (e.g. http://web.mit.edu/10.001/Web/Tips/Converge.htm) over the full long name that is cumbersome to write. Also I think an API with atol/rtol is less confusing than places/delta.